sdk/core/System.ClientModel/src/Convenience/ClientResultException.cs

```diff
@@ -96,6 +100,8 @@ private static async ValueTask<string> CreateMessageAsync(PipelineResponse respo
 
     private static async ValueTask<string> CreateMessageSyncOrAsync(PipelineResponse response, bool async)
     {
+        Argument.AssertNotNull(response, nameof(response));
```

Comment:
Do we need this one, if we're already validating on the public members?

Reply:
It actually gets called statically before the constructor ones are called, so we would throw NRE if we didn't validate here.  That means technically the public constructor ones aren't needed, but I wanted to keep them for clarity, and to cover all the paths consistently.